BeamSpot + Rho variables in nanoAODs

maxgalli opened the pull request:
#### PR description:

This PR introduces the photon variables required to allow H->gamma gamma to move to nanoAODs.
The changes were discussed in [this](https://indico.cern.ch/event/1085967/#4-hgg-photon-updates-for-nano) XPOG meeting.

### EDIT

The photons related variables were moved to https://github.com/cms-sw/cmssw/pull/36526

lgray committed:
add missing vars

maxgalli committed:
Add beamspot table and task

maxgalli committed:
Add int8 support

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-36037/26493

- This PR adds an extra 36KB to repository

- There are other open Pull requests which might conflict with changes you have proposed:
  - File  DataFormats/NanoAOD/src/classes_def.xml  modified in PR(s): #32167
  - File  PhysicsTools/NanoAOD/interface/SimpleFlatTableProducer.h  modified in PR(s): #34169
  - File  PhysicsTools/NanoAOD/plugins/SimpleFlatTableProducerPlugins.cc  modified in PR(s): #34169
  - File  PhysicsTools/NanoAOD/plugins/TableOutputBranches.cc  modified in PR(s): #32167
  - File  PhysicsTools/NanoAOD/plugins/TableOutputBranches.h  modified in PR(s): #32167
  - File  PhysicsTools/NanoAOD/python/globals_cff.py  modified in PR(s): #34169

cmsbuild commented:
A new Pull Request was created by  @maxgalli (Massimiliano Galli) for master.

It involves the following packages:

- DataFormats/NanoAOD (**xpog**)
- PhysicsTools/NanoAOD (**xpog**)


@cmsbuild, @mariadalfonso, @gouskos, @fgolf can you please review it and eventually sign? Thanks.
@gpetruc, @rovere, @swertz this is something you requested to watch as well.
@perrotta, @dpiparo, @qliphy you are the release manager for this.

cms-bot commands are listed <a href="http://cms-sw.github.io/cms-bot-cmssw-cmds.html">here</a>

mariadalfonso commented:
please test

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-36037/26495

- This PR adds an extra 40KB to repository

- There are other open Pull requests which might conflict with changes you have proposed:
  - File  DataFormats/NanoAOD/src/classes_def.xml  modified in PR(s): #32167
  - File  PhysicsTools/NanoAOD/interface/SimpleFlatTableProducer.h  modified in PR(s): #34169
  - File  PhysicsTools/NanoAOD/plugins/SimpleFlatTableProducerPlugins.cc  modified in PR(s): #34169
  - File  PhysicsTools/NanoAOD/plugins/TableOutputBranches.cc  modified in PR(s): #32167
  - File  PhysicsTools/NanoAOD/plugins/TableOutputBranches.h  modified in PR(s): #32167
  - File  PhysicsTools/NanoAOD/python/globals_cff.py  modified in PR(s): #34169

cmsbuild commented:
Pull request #36037 was updated. @cmsbuild, @mariadalfonso, @gouskos, @fgolf can you please check and sign again.

gouskos commented:
please test

cmsbuild commented:
-1

**Failed Tests**: RelVals 
**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-7fdc3b/20341/summary.html
**COMMIT**: 351ebb1307cdac07b9dbcf6dd3ab086a8c76db7b
**CMSSW**: CMSSW_12_2_X_2021-11-08-1100/slc7_amd64_gcc900
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week0/cms-sw/cmssw/36037/20341/install.sh` to create a dev area with all the needed externals and cmssw changes.


## RelVals

- [11634.911](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-7fdc3b/20341/runTheMatrix-results/11634.911_TTbar_14TeV+2021_DD4hep+TTbar_14TeV_TuneCP5_GenSim+Digi+Reco+HARVEST+ALCA)
```
----- Begin Fatal Exception 08-Nov-2021 18:07:45 CET-----------------------
An exception of category 'PluginNotFound' occurred while
   [0] Constructing the EventProcessor
Exception Message:
Unable to find plugin 'EcalBarrelGeometryEPdd4hep' in category 'CMS EDM Framework ParameterSet Description'. Please check spelling of name.
----- End Fatal Exception -------------------------------------------------
```

mariadalfonso commented:
please test

cmsbuild commented:
-1

**Failed Tests**: RelVals 
**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-7fdc3b/20398/summary.html
**COMMIT**: 351ebb1307cdac07b9dbcf6dd3ab086a8c76db7b
**CMSSW**: CMSSW_12_2_X_2021-11-08-2300/slc7_amd64_gcc900
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week0/cms-sw/cmssw/36037/20398/install.sh` to create a dev area with all the needed externals and cmssw changes.


## RelVals

- [11634.911](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-7fdc3b/20398/runTheMatrix-results/11634.911_TTbar_14TeV+2021_DD4hep+TTbar_14TeV_TuneCP5_GenSim+Digi+Reco+HARVEST+ALCA)
```
----- Begin Fatal Exception 09-Nov-2021 18:22:03 CET-----------------------
An exception of category 'PluginNotFound' occurred while
   [0] Constructing the EventProcessor
Exception Message:
Unable to find plugin 'EcalBarrelGeometryEPdd4hep' in category 'CMS EDM Framework ParameterSet Description'. Please check spelling of name.
----- End Fatal Exception -------------------------------------------------
```

maxgalli commented:
> @maxgalli we need to add the extra variables in the DQM PhysicsTools/NanoAOD/python/nanoDQM_cfi.py

ok, I will take care soon

cmsbuild commented:
+1

**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-7fdc3b/20546/summary.html
**COMMIT**: 351ebb1307cdac07b9dbcf6dd3ab086a8c76db7b
**CMSSW**: CMSSW_12_2_X_2021-11-15-1100/slc7_amd64_gcc900
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week1/cms-sw/cmssw/36037/20546/install.sh` to create a dev area with all the needed externals and cmssw changes.


## Comparison Summary



**Summary**:
 - No significant changes to the logs found
 - ROOTFileChecks: Some differences in event products or their sizes found
 - Reco comparison results: 138 differences found in the comparisons
 - DQMHistoTests: Total files compared: 42
 - DQMHistoTests: Total histograms compared: 3320802
 - DQMHistoTests: Total failures: 5
 - DQMHistoTests: Total nulls: 1
 - DQMHistoTests: Total successes: 3320774
 - DQMHistoTests: Total skipped: 22
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: -0.004 KiB( 41 files compared)
 - DQMHistoSizes: changed ( 312.0 ): -0.004 KiB       MessageLogger/Warnings
 - Checked 177 log files, 37 edm output root files, 42 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_12_2_X_2021-11-15-1100+7fdc3b/46866/triggerResults): no differences found

mariadalfonso commented:
> * Reco comparison results: 138 differences found in the comparisons

There are differences in the photon and rho tables, touches by this PR.
but also in muon, electron, tau , isoTrack, rivetLepton to be checked if are related to the addition of int in the DataFormat

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-36037/26658

- This PR adds an extra 24KB to repository

- There are other open Pull requests which might conflict with changes you have proposed:
  - File  DataFormats/NanoAOD/src/classes_def.xml  modified in PR(s): #32167
  - File  PhysicsTools/NanoAOD/interface/SimpleFlatTableProducer.h  modified in PR(s): #34169
  - File  PhysicsTools/NanoAOD/plugins/SimpleFlatTableProducerPlugins.cc  modified in PR(s): #34169
  - File  PhysicsTools/NanoAOD/plugins/TableOutputBranches.cc  modified in PR(s): #32167
  - File  PhysicsTools/NanoAOD/plugins/TableOutputBranches.h  modified in PR(s): #32167
  - File  PhysicsTools/NanoAOD/python/globals_cff.py  modified in PR(s): #34169
  - File  PhysicsTools/NanoAOD/python/nanoDQM_cfi.py  modified in PR(s): #35872

cmsbuild commented:
Pull request #36037 was updated. @cmsbuild, @mariadalfonso, @gouskos, @fgolf can you please check and sign again.

mariadalfonso commented:
can you add also the rho and beamSpot ? Thanks

maxgalli committed:
Add rho and beamspot variables to DQM

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-36037/26685

- This PR adds an extra 52KB to repository

- There are other open Pull requests which might conflict with changes you have proposed:
  - File  DataFormats/NanoAOD/src/classes_def.xml  modified in PR(s): #32167
  - File  PhysicsTools/NanoAOD/interface/SimpleFlatTableProducer.h  modified in PR(s): #34169
  - File  PhysicsTools/NanoAOD/plugins/SimpleFlatTableProducerPlugins.cc  modified in PR(s): #34169
  - File  PhysicsTools/NanoAOD/plugins/TableOutputBranches.cc  modified in PR(s): #32167
  - File  PhysicsTools/NanoAOD/plugins/TableOutputBranches.h  modified in PR(s): #32167
  - File  PhysicsTools/NanoAOD/python/globals_cff.py  modified in PR(s): #34169
  - File  PhysicsTools/NanoAOD/python/nanoDQM_cfi.py  modified in PR(s): #35872

cmsbuild commented:
Pull request #36037 was updated. @cmsbuild, @mariadalfonso, @gouskos, @fgolf can you please check and sign again.

gouskos commented:
please test

cmsbuild commented:
-1

**Failed Tests**: RelVals RelVals-INPUT 
**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-7fdc3b/20585/summary.html
**COMMIT**: 1e6da796436831052bd9966b8463995005351904
**CMSSW**: CMSSW_12_2_X_2021-11-16-2300/slc7_amd64_gcc900
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week1/cms-sw/cmssw/36037/20585/install.sh` to create a dev area with all the needed externals and cmssw changes.


## RelVals

- [136.8311](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-7fdc3b/20585/runTheMatrix-results/136.8311_RunJetHT2017F_reminiaod+RunJetHT2017F_reminiaod+REMINIAOD_data2017+HARVEST2017_REMINIAOD_data2017)```136.8311_RunJetHT2017F_reminiaod+RunJetHT2017F_reminiaod+REMINIAOD_data2017+HARVEST2017_REMINIAOD_data2017/step2_RunJetHT2017F_reminiaod+RunJetHT2017F_reminiaod+REMINIAOD_data2017+HARVEST2017_REMINIAOD_data2017.log```
- [136.88811](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-7fdc3b/20585/runTheMatrix-results/136.88811_RunJetHT2018D_reminiaodUL+RunJetHT2018D_reminiaodUL+REMINIAOD_data2018UL+HARVEST2018_REMINIAOD_data2018UL)```136.88811_RunJetHT2018D_reminiaodUL+RunJetHT2018D_reminiaodUL+REMINIAOD_data2018UL+HARVEST2018_REMINIAOD_data2018UL/step2_RunJetHT2018D_reminiaodUL+RunJetHT2018D_reminiaodUL+REMINIAOD_data2018UL+HARVEST2018_REMINIAOD_data2018UL.log```
- [136.8523](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-7fdc3b/20585/runTheMatrix-results/136.8523_RunJetHT2018C_nanoULremini+RunJetHT2018C_nanoULremini+NANOEDM2018_106Xv2+HARVESTNANOAOD2018_106Xv2)```136.8523_RunJetHT2018C_nanoULremini+RunJetHT2018C_nanoULremini+NANOEDM2018_106Xv2+HARVESTNANOAOD2018_106Xv2/step2_RunJetHT2018C_nanoULremini+RunJetHT2018C_nanoULremini+NANOEDM2018_106Xv2+HARVESTNANOAOD2018_106Xv2.log```
<details>
<summary>Expand to see more relval errors ...</summary>

- [136.7611](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-7fdc3b/20585/runTheMatrix-results/136.7611_RunJetHT2016E_reminiaod+RunJetHT2016E_reminiaod+REMINIAOD_data2016_HIPM+HARVESTDR2_REMINIAOD_data2016_HIPM)
- [1325.81](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-7fdc3b/20585/runTheMatrix-results/1325.81_TTbar_13_106Xv1NanoAODINPUT+TTbar_13_106Xv1NanoAODINPUT+NANOEDMMC2017_106XMiniAODv1+HARVESTNANOAODMC2017_106XMiniAODv1)
- [158.01](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-7fdc3b/20585/runTheMatrix-results/158.01_HydjetQ_reminiaodPbPb2018_INPUT+HydjetQ_reminiaodPbPb2018_INPUT+REMINIAODHI2018PPRECOMB+HARVESTHI2018PPRECOMINIAOD)
- [140.56](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-7fdc3b/20585/runTheMatrix-results/140.56_RunHI2018+RunHI2018+RECOHID18+HARVESTDHI18)
- [4.53](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-7fdc3b/20585/runTheMatrix-results/4.53_RunPhoton2012B+RunPhoton2012B+HLTD+RECODR1reHLT+HARVESTDR1reHLT)
- [136.731](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-7fdc3b/20585/runTheMatrix-results/136.731_RunSinglePh2016B+RunSinglePh2016B+HLTDR2_2016+RECODR2_2016reHLT_skimSinglePh_HIPM+HARVESTDR2)
- [136.874](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-7fdc3b/20585/runTheMatrix-results/136.874_RunEGamma2018C+RunEGamma2018C+HLTDR2_2018+RECODR2_2018reHLT_skimEGamma_Offline_L1TEgDQM+HARVEST2018_L1TEgDQM)
- [136.793](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-7fdc3b/20585/runTheMatrix-results/136.793_RunDoubleEG2017C+RunDoubleEG2017C+HLTDR2_2017+RECODR2_2017reHLT_skimDoubleEG_Prompt+HARVEST2017)
- [1000.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-7fdc3b/20585/runTheMatrix-results/1000.0_RunMinBias2011A+RunMinBias2011A+TIER0+SKIMD+HARVESTDfst2+ALCASPLIT)
- [1001.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-7fdc3b/20585/runTheMatrix-results/1001.0_RunMinBias2011A+RunMinBias2011A+TIER0EXP+ALCAEXP+ALCAHARVDSIPIXELCALRUN1+ALCAHARVD1+ALCAHARVD2+ALCAHARVD3+ALCAHARVD4+ALCAHARVD5+ALCAHARVD7)
- [135.4](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-7fdc3b/20585/runTheMatrix-results/135.4_ZEE_13+ZEEFS_13+HARVESTUP15FS+MINIAODMCUP15FS)
- [5.1](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-7fdc3b/20585/runTheMatrix-results/5.1_TTbar+TTbarFS+HARVESTFS)
- [1306.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-7fdc3b/20585/runTheMatrix-results/1306.0_SingleMuPt1_UP15+SingleMuPt1_UP15+DIGIUP15+RECOUP15+HARVESTUP15)
- [2018.1](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-7fdc3b/20585/runTheMatrix-results/2018.1_TTbar_13_UP18+TTbarFS_13_UP18+HARVESTUP18FS+MINIAODMCUP18FS)
- [1330.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-7fdc3b/20585/runTheMatrix-results/1330.0_ZMM_13+ZMM_13+DIGIUP15+RECOUP15_L1TMuDQM+HARVESTUP15_L1TMuDQM+NANOUP15)
- [312.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-7fdc3b/20585/runTheMatrix-results/312.0_Pyquen_ZeemumuJets_pt10_2760GeV_2021+Pyquen_ZeemumuJets_pt10_2760GeV_2021+DIGIHI2021MIX+RECOHI2021MIX+HARVESTHI2021PPRECO)
- [9.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-7fdc3b/20585/runTheMatrix-results/9.0_Higgs200ChargedTaus+Higgs200ChargedTaus+DIGI+RECO+HARVEST)
- [25.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-7fdc3b/20585/runTheMatrix-results/25.0_TTbar+TTbar+DIGI+RECOAlCaCalo+HARVEST+ALCATT)
- [10042.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-7fdc3b/20585/runTheMatrix-results/10042.0_ZMM_13+2017+ZMM_13TeV_TuneCUETP8M1_GenSim+Digi+RecoFakeHLT+HARVESTFakeHLT+ALCA+Nano)
- [10024.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-7fdc3b/20585/runTheMatrix-results/10024.0_TTbar_13+2017+TTbar_13TeV_TuneCUETP8M1_GenSim+Digi+RecoFakeHLT+HARVESTFakeHLT+ALCA+Nano)
- [10824.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-7fdc3b/20585/runTheMatrix-results/10824.0_TTbar_13+2018+TTbar_13TeV_TuneCUETP8M1_GenSim+Digi+RecoFakeHLT+HARVESTFakeHLT+ALCA+Nano)
- [11634.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-7fdc3b/20585/runTheMatrix-results/11634.0_TTbar_14TeV+2021+TTbar_14TeV_TuneCP5_GenSim+Digi+RecNan+HARVESTRecNan+ALCA)
- [11634.7](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-7fdc3b/20585/runTheMatrix-results/11634.7_TTbar_14TeV+2021_trackingMkFit+TTbar_14TeV_TuneCP5_GenSim+Digi+Reco+HARVEST)
- [11634.914](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-7fdc3b/20585/runTheMatrix-results/11634.914_TTbar_14TeV+2021_DDDDB+TTbar_14TeV_TuneCP5_GenSim+Digi+Reco+HARVEST+ALCA)
- [25202.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-7fdc3b/20585/runTheMatrix-results/25202.0_TTbar_13+TTbar_13+DIGIUP15_PU25+RECOUP15_PU25+HARVESTUP15_PU25+NANOUP15_PU25)
- [11634.911](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-7fdc3b/20585/runTheMatrix-results/11634.911_TTbar_14TeV+2021_DD4hep+TTbar_14TeV_TuneCP5_GenSim+Digi+Reco+HARVEST+ALCA)
- [10224.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-7fdc3b/20585/runTheMatrix-results/10224.0_TTbar_13+2017PU+TTbar_13TeV_TuneCUETP8M1_GenSim+DigiPU+RecoFakeHLTPU+HARVESTFakeHLTPU+Nano)
- [12434.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-7fdc3b/20585/runTheMatrix-results/12434.0_TTbar_14TeV+2023+TTbar_14TeV_TuneCP5_GenSim+Digi+RecNan+HARVESTRecNan+ALCA)
- [38634.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-7fdc3b/20585/runTheMatrix-results/38634.0_TTbar_14TeV+2026D86+TTbar_14TeV_TuneCP5_GenSimHLBeamSpot14+DigiTrigger+RecoGlobal+HARVESTGlobal)
- [23234.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-7fdc3b/20585/runTheMatrix-results/23234.0_TTbar_14TeV+2026D49+TTbar_14TeV_TuneCP5_GenSimHLBeamSpot14+DigiTrigger+RecoGlobal+HARVESTGlobal)
- [35034.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-7fdc3b/20585/runTheMatrix-results/35034.0_TTbar_14TeV+2026D77+TTbar_14TeV_TuneCP5_GenSimHLBeamSpot14+DigiTrigger+RecoGlobal+HARVESTGlobal)
- [28234.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-7fdc3b/20585/runTheMatrix-results/28234.0_TTbar_14TeV+2026D60+TTbar_14TeV_TuneCP5_GenSimHLBeamSpot14+DigiTrigger+RecoGlobal+HARVESTGlobal)
- [250202.181](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-7fdc3b/20585/runTheMatrix-results/250202.181_TTbar_13UP18+TTbar_13UP18+PREMIXUP18_PU25+DIGIPRMXLOCALUP18_PU25+RECOPRMXUP18_PU25+HARVESTUP18_PU25)
- [35234.999](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-7fdc3b/20585/runTheMatrix-results/35234.999_TTbar_14TeV+2026D77PU_PMXS1S2PR+TTbar_14TeV_TuneCP5_GenSimHLBeamSpot14+PREMIX_PremixHLBeamSpot14PU+DigiTriggerPU+RecoGlobalPU+HARVESTGlobalPU)
</details>



## RelVals-INPUT

- [4.6](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-7fdc3b/20585/runTheMatrixINPUT-results/4.6_MinimumBias2010A+MinimumBias2010A+RECOSKIMALCA+HARVESTDR1)```4.6_MinimumBias2010A+MinimumBias2010A+RECOSKIMALCA+HARVESTDR1/step2_MinimumBias2010A+MinimumBias2010A+RECOSKIMALCA+HARVESTDR1.log```
- [136.72411](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-7fdc3b/20585/runTheMatrixINPUT-results/136.72411_RunJetHT2016B_reminiaodUL+RunJetHT2016B_reminiaodUL+REMINIAOD_data2016UL_HIPM+HARVESTDR2_REMINIAOD_data2016UL_HIPM)```136.72411_RunJetHT2016B_reminiaodUL+RunJetHT2016B_reminiaodUL+REMINIAOD_data2016UL_HIPM+HARVESTDR2_REMINIAOD_data2016UL_HIPM/step2_RunJetHT2016B_reminiaodUL+RunJetHT2016B_reminiaodUL+REMINIAOD_data2016UL_HIPM+HARVESTDR2_REMINIAOD_data2016UL_HIPM.log```
- [136.72412](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-7fdc3b/20585/runTheMatrixINPUT-results/136.72412_RunJetHT2016B_reminiaodUL+RunJetHT2016B_reminiaodUL+REMININANO_data2016UL_HIPM+HARVESTDR2_REMININANO_data2016UL_HIPM)```136.72412_RunJetHT2016B_reminiaodUL+RunJetHT2016B_reminiaodUL+REMININANO_data2016UL_HIPM+HARVESTDR2_REMININANO_data2016UL_HIPM/step2_RunJetHT2016B_reminiaodUL+RunJetHT2016B_reminiaodUL+REMININANO_data2016UL_HIPM+HARVESTDR2_REMININANO_data2016UL_HIPM.log```
<details>
<summary>Expand to see more relval errors ...</summary>

- [136.7611](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-7fdc3b/20585/runTheMatrixINPUT-results/136.7611_RunJetHT2016E_reminiaod+RunJetHT2016E_reminiaod+REMINIAOD_data2016_HIPM+HARVESTDR2_REMINIAOD_data2016_HIPM)
- [136.76111](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-7fdc3b/20585/runTheMatrixINPUT-results/136.76111_RunJetHT2016E_reminiaodUL+RunJetHT2016E_reminiaodUL+REMINIAOD_data2016UL_HIPM+HARVESTDR2_REMINIAOD_data2016UL_HIPM)
- [136.7722](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-7fdc3b/20585/runTheMatrixINPUT-results/136.7722_RunJetHT2016H_nano+RunJetHT2016H_nano+NANOEDM2016_80X+HARVESTNANOAOD2016_80X)
- [136.77211](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-7fdc3b/20585/runTheMatrixINPUT-results/136.77211_RunJetHT2016H_reminiaodUL+RunJetHT2016H_reminiaodUL+REMINIAOD_data2016UL+HARVESTDR2_REMINIAOD_data2016UL)
- [136.7721](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-7fdc3b/20585/runTheMatrixINPUT-results/136.7721_RunJetHT2016H_reminiaod+RunJetHT2016H_reminiaod+REMINIAOD_data2016+HARVESTDR2_REMINIAOD_data2016)
- [136.7802](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-7fdc3b/20585/runTheMatrixINPUT-results/136.7802_RunHLTPhy2017B_AODextra+RunHLTPhy2017B_AODextra+DQMHLTonAODextra_2017+HARVESTDQMHLTonAOD_2017)
- [136.7801](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-7fdc3b/20585/runTheMatrixINPUT-results/136.7801_RunHLTPhy2017B_AOD+RunHLTPhy2017B_AOD+DQMHLTonAOD_2017+HARVESTDQMHLTonAOD_2017)
- [136.7803](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-7fdc3b/20585/runTheMatrixINPUT-results/136.7803_RunHLTPhy2017B_RAWAOD+RunHLTPhy2017B_RAWAOD+DQMHLTonRAWAOD_2017+HARVESTDQMHLTonAOD_2017)
- [136.7952](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-7fdc3b/20585/runTheMatrixINPUT-results/136.7952_RunJetHT2017C_94Xv2NanoAODINPUT+RunJetHT2017C_94Xv2NanoAODINPUT+NANOEDM2017_94XMiniAODv2+HARVESTNANOAOD2017_94XMiniAODv2)
- [136.83111](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-7fdc3b/20585/runTheMatrixINPUT-results/136.83111_RunJetHT2017F_reminiaodUL+RunJetHT2017F_reminiaodUL+REMINIAOD_data2017UL+HARVEST2017_REMINIAOD_data2017UL)
- [136.8311](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-7fdc3b/20585/runTheMatrixINPUT-results/136.8311_RunJetHT2017F_reminiaod+RunJetHT2017F_reminiaod+REMINIAOD_data2017+HARVEST2017_REMINIAOD_data2017)
- [136.8523](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-7fdc3b/20585/runTheMatrixINPUT-results/136.8523_RunJetHT2018C_nanoULremini+RunJetHT2018C_nanoULremini+NANOEDM2018_106Xv2+HARVESTNANOAOD2018_106Xv2)
- [136.8521](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-7fdc3b/20585/runTheMatrixINPUT-results/136.8521_RunJetHT2018A_nano+RunJetHT2018A_nano+NANOEDM2018_102Xv1+HARVESTNANOAOD2018_102Xv1)
- [136.8522](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-7fdc3b/20585/runTheMatrixINPUT-results/136.8522_RunJetHT2018A_nanoUL+RunJetHT2018A_nanoUL+NANOEDM2018_106Xv1+HARVESTNANOAOD2018_106Xv1)
- [136.88811](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-7fdc3b/20585/runTheMatrixINPUT-results/136.88811_RunJetHT2018D_reminiaodUL+RunJetHT2018D_reminiaodUL+REMINIAOD_data2018UL+HARVEST2018_REMINIAOD_data2018UL)
- [140.5611](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-7fdc3b/20585/runTheMatrixINPUT-results/140.5611_RunHI2018AOD+RunHI2018AOD+REMINIAODHID18+HARVESTHI18MINIAOD)
- [158.01](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-7fdc3b/20585/runTheMatrixINPUT-results/158.01_HydjetQ_reminiaodPbPb2018_INPUT+HydjetQ_reminiaodPbPb2018_INPUT+REMINIAODHI2018PPRECOMB+HARVESTHI2018PPRECOMINIAOD)
- [140.54](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-7fdc3b/20585/runTheMatrixINPUT-results/140.54_RunPA2013+RunPA2013+RECO_PPbData+HARVEST_PPbData)
- [140.56](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-7fdc3b/20585/runTheMatrixINPUT-results/140.56_RunHI2018+RunHI2018+RECOHID18+HARVESTDHI18)
- [140.57](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-7fdc3b/20585/runTheMatrixINPUT-results/140.57_RunHI2018Reduced+RunHI2018Reduced+RECOHID18+HARVESTDHI18)
- [1325.5](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-7fdc3b/20585/runTheMatrixINPUT-results/1325.5_ProdZEE_13_reminiaodINPUT+ProdZEE_13_reminiaodINPUT+REMINIAOD_mc2016+HARVESTDR2_REMINIAOD_mc2016)
- [1325.5161](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-7fdc3b/20585/runTheMatrixINPUT-results/1325.5161_TTbar_13_reminiaod2016UL_postVFP_INPUT+TTbar_13_reminiaod2016UL_postVFP_INPUT+REMINIAOD_mc2016UL_postVFP+HARVESTDR2_REMINIAOD_mc2016UL_postVFP)
- [1325.51](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-7fdc3b/20585/runTheMatrixINPUT-results/1325.51_TTbar_13_94XreminiaodINPUT+TTbar_13_94XreminiaodINPUT+REMINIAOD_mc2017+HARVESTUP17_REMINIAOD_mc2017)
- [1325.517](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-7fdc3b/20585/runTheMatrixINPUT-results/1325.517_TTbar_13_reminiaod2017UL_INPUT+TTbar_13_reminiaod2017UL_INPUT+REMINIAOD_mc2017UL+HARVESTUP17_REMINIAOD_mc2017UL)
- [1325.516](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-7fdc3b/20585/runTheMatrixINPUT-results/1325.516_TTbar_13_reminiaod2016UL_preVFP_INPUT+TTbar_13_reminiaod2016UL_preVFP_INPUT+REMINIAOD_mc2016UL_preVFP+HARVESTDR2_REMINIAOD_mc2016UL_preVFP)
- [1325.6](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-7fdc3b/20585/runTheMatrixINPUT-results/1325.6_TTbar_13_94Xv1NanoAODINPUT+TTbar_13_94Xv1NanoAODINPUT+NANOAODMC2017_94XMiniAODv1)
- [1325.61](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-7fdc3b/20585/runTheMatrixINPUT-results/1325.61_TTbar_13_106Xv1NanoAODINPUT+TTbar_13_106Xv1NanoAODINPUT+NANOAODMC2017_106XMiniAODv1)
- [1325.7](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-7fdc3b/20585/runTheMatrixINPUT-results/1325.7_TTbar_13_94Xv2NanoAODINPUT+TTbar_13_94Xv2NanoAODINPUT+NANOEDMMC2017_94XMiniAODv2+HARVESTNANOAODMC2017_94XMiniAODv2)
- [1325.8](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-7fdc3b/20585/runTheMatrixINPUT-results/1325.8_TTbar_13_94Xv1NanoAODINPUT+TTbar_13_94Xv1NanoAODINPUT+NANOEDMMC2017_94XMiniAODv1+HARVESTNANOAODMC2017_94XMiniAODv1)
- [1325.518](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-7fdc3b/20585/runTheMatrixINPUT-results/1325.518_TTbar_13_reminiaod2018UL_INPUT+TTbar_13_reminiaod2018UL_INPUT+REMINIAOD_mc2018UL+HARVESTUP18_REMINIAOD_mc2018UL)
- [1325.81](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-7fdc3b/20585/runTheMatrixINPUT-results/1325.81_TTbar_13_106Xv1NanoAODINPUT+TTbar_13_106Xv1NanoAODINPUT+NANOEDMMC2017_106XMiniAODv1+HARVESTNANOAODMC2017_106XMiniAODv1)
- [1329.1](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-7fdc3b/20585/runTheMatrixINPUT-results/1329.1_ZEE_13_80XNanoAODINPUT+ZEE_13_80XNanoAODINPUT+NANOEDMMC2016_80X+HARVESTNANOAODMC2016_80X)
- [134.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-7fdc3b/20585/runTheMatrixINPUT-results/134.0_MinimumBiasBS+RunMinBias2011B+RECODR1+ALCAPROMPT+ALCAHARVD)
- [144.6](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-7fdc3b/20585/runTheMatrixINPUT-results/144.6_MinimumBias2010A+MinimumBias2010A+RECOSKIMALCAR1)
- [1000.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-7fdc3b/20585/runTheMatrixINPUT-results/1000.0_RunMinBias2011A+RunMinBias2011A+TIER0+SKIMD+HARVESTDfst2+ALCASPLIT)
- [1001.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-7fdc3b/20585/runTheMatrixINPUT-results/1001.0_RunMinBias2011A+RunMinBias2011A+TIER0EXP+ALCAEXP+ALCAHARVDSIPIXELCALRUN1+ALCAHARVD1+ALCAHARVD2+ALCAHARVD3+ALCAHARVD4+ALCAHARVD5+ALCAHARVD7)
- [1003.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-7fdc3b/20585/runTheMatrixINPUT-results/1003.0_RunMinBias2012A+RunMinBias2012A+RECODDQM+HARVESTDDQM)
- [1001.2](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-7fdc3b/20585/runTheMatrixINPUT-results/1001.2_RunZeroBias2017F+RunZeroBias2017F+TIER0EXPRUN2+ALCAEXPRUN2+ALCAHARVDSIPIXELCAL)
- [1002.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-7fdc3b/20585/runTheMatrixINPUT-results/1002.0_RRD+RunMinBias2011A+RECODR1+COPYPASTE)
- [1030.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-7fdc3b/20585/runTheMatrixINPUT-results/1030.0_RunHLTPhy2017B+RunHLTPhy2017B+TIER0EXPHPBS+ALCASPLITHPBS+ALCAHARVDHPBS+ALCAHARVDHPBSLOWPU)
- [1040.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-7fdc3b/20585/runTheMatrixINPUT-results/1040.0_RunZeroBias2017F+RunZeroBias2017F+TIER0RAWSIPIXELCAL+ALCASPLITSIPIXELCAL+ALCAHARVDSIPIXELCAL)
</details>

maxgalli commented:
> /cvmfs/cms-ci.cern.ch/week1/cms-sw/cmssw/36037/20585/install.sh

most likely due to the way I added variables to the DQM for rho and beamspot, trying to fix it now

maxgalli commented:
> > /cvmfs/cms-ci.cern.ch/week1/cms-sw/cmssw/36037/20585/install.sh
> 
> most likely due to the way I added variables to the DQM for rho and beamspot, trying to fix it now

I actually forgot the title key in two of the histograms, fixed now

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-36037/26701

- This PR adds an extra 28KB to repository

- There are other open Pull requests which might conflict with changes you have proposed:
  - File  DataFormats/NanoAOD/src/classes_def.xml  modified in PR(s): #32167
  - File  PhysicsTools/NanoAOD/interface/SimpleFlatTableProducer.h  modified in PR(s): #34169
  - File  PhysicsTools/NanoAOD/plugins/SimpleFlatTableProducerPlugins.cc  modified in PR(s): #34169
  - File  PhysicsTools/NanoAOD/plugins/TableOutputBranches.cc  modified in PR(s): #32167
  - File  PhysicsTools/NanoAOD/plugins/TableOutputBranches.h  modified in PR(s): #32167
  - File  PhysicsTools/NanoAOD/python/globals_cff.py  modified in PR(s): #34169
  - File  PhysicsTools/NanoAOD/python/nanoDQM_cfi.py  modified in PR(s): #35872

cmsbuild commented:
Pull request #36037 was updated. @cmsbuild, @mariadalfonso, @gouskos, @fgolf can you please check and sign again.

mariadalfonso commented:
please test

cmsbuild commented:
+1

**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-7fdc3b/20598/summary.html
**COMMIT**: f0e3be3cbad003748b88f1b0ec5d3c84868c910c
**CMSSW**: CMSSW_12_2_X_2021-11-17-1100/slc7_amd64_gcc900
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week1/cms-sw/cmssw/36037/20598/install.sh` to create a dev area with all the needed externals and cmssw changes.


## Comparison Summary



**Summary**:
 - No significant changes to the logs found
 - ROOTFileChecks: Some differences in event products or their sizes found
 - Reco comparison results: 138 differences found in the comparisons
 - DQMHistoTests: Total files compared: 42
 - DQMHistoTests: Total histograms compared: 3327156
 - DQMHistoTests: Total failures: 5
 - DQMHistoTests: Total nulls: 1
 - DQMHistoTests: Total successes: 3327128
 - DQMHistoTests: Total skipped: 22
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 48.374 KiB( 41 files compared)
 - DQMHistoSizes: changed ( 11634.0,... ): 13.749 KiB       Physics/NanoAODDQM
 - DQMHistoSizes: changed ( 136.8523 ): 7.131 KiB        Physics/NanoAODDQM
 - DQMHistoSizes: changed ( 312.0 ): -0.004 KiB       MessageLogger/Warnings
 - Checked 177 log files, 37 edm output root files, 42 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_12_2_X_2021-11-17-1100+7fdc3b/46913/triggerResults): no differences found

mariadalfonso commented:
The range of the beam spot DQM is not well defined
<img width="1321" alt="Screen Shot 2021-11-26 at 14 56 02" src="https://user-images.githubusercontent.com/5656714/143591729-be22f584-8abf-435e-8be6-925a3ba229ac.png">

mariadalfonso commented:
Rho Changed from 5 separate float to a block with 6 floats

<img width="968" alt="Screen Shot 2021-11-26 at 15 30 57" src="https://user-images.githubusercontent.com/5656714/143596077-352d44b2-2520-4774-9812-ccedede2a432.png">

<img width="867" alt="image" src="https://user-images.githubusercontent.com/5656714/143596220-cb9d5138-2fad-4c64-8797-a48898b0d532.png">

mariadalfonso commented:
There are numerous changes for the non photons/Rho/BeamSpot listed here 
https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_12_2_X_2021-11-17-1100+7fdc3b/46913/validateJR/all_OldVSNew_RunJetHT2018CnanoEDM106Xv2wf136p8523/

Among them: (goes in the right direction to remove these  UChar_t )
NEW Electron_cutBased       Int_t   0.4     1.5             1.0
WAS Electron_lostHits      UChar_t 0.4     1.5             1.0%

once can see the dump here:

this PR
https://cms-nanoaod-integration.web.cern.ch/integration/36037/mc106Xul17v2_size_report.html
vs the reference 
https://cms-nanoaod-integration.web.cern.ch/integration/cms-swmaster/mc106Xul17v2_size_report.html

mariadalfonso commented:
@maxgalli

This PR has conflict now.

To proceed with this PR we can do the following: move in a separate PR the photon changes (this add 8% of size so we should optimise separately) and keep this PR for rho and beamspot and related changes in Output class ( that is used for the beamspot)

in parallel: resolve conflict, fix ranges for DQM and of course understand why we have these unwanted changes https://github.com/cms-sw/cmssw/pull/36037#issuecomment-980024682

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-36037/27428

- This PR adds an extra 40KB to repository

- There are other open Pull requests which might conflict with changes you have proposed:
  - File  DataFormats/NanoAOD/src/classes_def.xml  modified in PR(s): #32167
  - File  PhysicsTools/NanoAOD/plugins/TableOutputBranches.cc  modified in PR(s): #32167
  - File  PhysicsTools/NanoAOD/plugins/TableOutputBranches.h  modified in PR(s): #32167
  - File  PhysicsTools/NanoAOD/python/nanoDQM_cfi.py  modified in PR(s): #36526
  - File  PhysicsTools/NanoAOD/python/photons_cff.py  modified in PR(s): #36526

cmsbuild commented:
Pull request #36037 was updated. @cmsbuild, @mariadalfonso, @gouskos, @fgolf can you please check and sign again.

mariadalfonso commented:
please test

cmsbuild commented:
+1

**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-7fdc3b/21335/summary.html
**COMMIT**: 58d012643f2e32439c816dd97e6266071b01229e
**CMSSW**: CMSSW_12_3_X_2021-12-16-1100/slc7_amd64_gcc10
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week1/cms-sw/cmssw/36037/21335/install.sh` to create a dev area with all the needed externals and cmssw changes.


## Comparison Summary



**Summary**:
 - No significant changes to the logs found
 - ROOTFileChecks: Some differences in event products or their sizes found
 - Reco comparison results: 5706 differences found in the comparisons
 - DQMHistoTests: Total files compared: 42
 - DQMHistoTests: Total histograms compared: 3250719
 - DQMHistoTests: Total failures: 7724
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 3242973
 - DQMHistoTests: Total skipped: 22
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 6.384 KiB( 41 files compared)
 - DQMHistoSizes: changed ( 11634.0,... ): 0.912 KiB        Physics/NanoAODDQM
 - Checked 177 log files, 37 edm output root files, 42 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_12_3_X_2021-12-16-1100+7fdc3b/47536/triggerResults): found differences in 2 / 41 workflows

mariadalfonso commented:
> * Reco comparison results: 5706 differences found in the comparisons

we have now unrelated changes also in MINI-AOD
will retest later with an extra IB

mariadalfonso commented:
please test

cmsbuild commented:
+1

**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-7fdc3b/21429/summary.html
**COMMIT**: 58d012643f2e32439c816dd97e6266071b01229e
**CMSSW**: CMSSW_12_3_X_2021-12-21-2300/slc7_amd64_gcc10
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week0/cms-sw/cmssw/36037/21429/install.sh` to create a dev area with all the needed externals and cmssw changes.


## Comparison Summary



**Summary**:
 - No significant changes to the logs found
 - ROOTFileChecks: Some differences in event products or their sizes found
 - Reco comparison results: 84 differences found in the comparisons
 - DQMHistoTests: Total files compared: 43
 - DQMHistoTests: Total histograms compared: 3461692
 - DQMHistoTests: Total failures: 5
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 3461665
 - DQMHistoTests: Total skipped: 22
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 7.296 KiB( 42 files compared)
 - DQMHistoSizes: changed ( 11634.0,... ): 0.912 KiB        Physics/NanoAODDQM
 - Checked 181 log files, 42 edm output root files, 43 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_12_3_X_2021-12-21-2300+7fdc3b/47621/triggerResults): no differences found

mariadalfonso commented:
please test

cmsbuild commented:
+1

**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-7fdc3b/21629/summary.html
**COMMIT**: 58d012643f2e32439c816dd97e6266071b01229e
**CMSSW**: CMSSW_12_3_X_2022-01-10-2300/slc7_amd64_gcc10
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week1/cms-sw/cmssw/36037/21629/install.sh` to create a dev area with all the needed externals and cmssw changes.


## Comparison Summary



**Summary**:
 - No significant changes to the logs found
 - ROOTFileChecks: Some differences in event products or their sizes found
 - Reco comparison results: 89 differences found in the comparisons
 - DQMHistoTests: Total files compared: 43
 - DQMHistoTests: Total histograms compared: 3461659
 - DQMHistoTests: Total failures: 10
 - DQMHistoTests: Total nulls: 1
 - DQMHistoTests: Total successes: 3461626
 - DQMHistoTests: Total skipped: 22
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 7.292 KiB( 42 files compared)
 - DQMHistoSizes: changed ( 11634.0,... ): 0.912 KiB        Physics/NanoAODDQM
 - DQMHistoSizes: changed ( 312.0 ): -0.004 KiB       MessageLogger/Warnings
 - Checked 181 log files, 42 edm output root files, 43 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_12_3_X_2022-01-10-2300+7fdc3b/47782/triggerResults): no differences found

gouskos commented:
please test

cmsbuild commented:
-1

**Failed Tests**: RelVals-INPUT 
**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-7fdc3b/21964/summary.html
**COMMIT**: 58d012643f2e32439c816dd97e6266071b01229e
**CMSSW**: CMSSW_12_3_X_2022-01-24-1100/slc7_amd64_gcc10
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week1/cms-sw/cmssw/36037/21964/install.sh` to create a dev area with all the needed externals and cmssw changes.


## RelVals-INPUT

- [4.76](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-7fdc3b/21964/runTheMatrixINPUT-results/4.76_ZMuSkim2012D+ZMuSkim2012D+HLTDSKIM2+RECODR1reHLT2+HARVESTDR1reHLT)```4.76_ZMuSkim2012D+ZMuSkim2012D+HLTDSKIM2+RECODR1reHLT2+HARVESTDR1reHLT/step2_ZMuSkim2012D+ZMuSkim2012D+HLTDSKIM2+RECODR1reHLT2+HARVESTDR1reHLT.log```


## Comparison Summary



**Summary**:
 - No significant changes to the logs found
 - ROOTFileChecks: Some differences in event products or their sizes found
 - Reco comparison results: 80 differences found in the comparisons
 - DQMHistoTests: Total files compared: 43
 - DQMHistoTests: Total histograms compared: 3449324
 - DQMHistoTests: Total failures: 0
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 3449302
 - DQMHistoTests: Total skipped: 22
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 7.296 KiB( 42 files compared)
 - DQMHistoSizes: changed ( 11634.0,... ): 0.912 KiB        Physics/NanoAODDQM
 - Checked 181 log files, 42 edm output root files, 43 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_12_3_X_2022-01-24-1100+7fdc3b/48026/triggerResults): no differences found

mariadalfonso commented on DataFormats/NanoAOD/interface/FlatTable.h:
```diff
@@ -41,6 +41,7 @@ namespace nanoaod {
       Float,
       Int,
       UInt8,
+      Int8,
```
can you have this as last of the enum ?
the changes in the histogram of type() seen by the jenkin might be simply related to a shift here

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-36037/28263

- This PR adds an extra 24KB to repository

- There are other open Pull requests which might conflict with changes you have proposed:
  - File  DataFormats/NanoAOD/src/classes_def.xml  modified in PR(s): #32167
  - File  PhysicsTools/NanoAOD/plugins/TableOutputBranches.cc  modified in PR(s): #32167
  - File  PhysicsTools/NanoAOD/plugins/TableOutputBranches.h  modified in PR(s): #32167
  - File  PhysicsTools/NanoAOD/python/nanoDQM_cfi.py  modified in PR(s): #36526
  - File  PhysicsTools/NanoAOD/python/photons_cff.py  modified in PR(s): #36845, #36526

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-36037/28264

- This PR adds an extra 24KB to repository

- There are other open Pull requests which might conflict with changes you have proposed:
  - File  DataFormats/NanoAOD/src/classes_def.xml  modified in PR(s): #32167
  - File  PhysicsTools/NanoAOD/plugins/TableOutputBranches.cc  modified in PR(s): #32167
  - File  PhysicsTools/NanoAOD/plugins/TableOutputBranches.h  modified in PR(s): #32167
  - File  PhysicsTools/NanoAOD/python/nanoDQM_cfi.py  modified in PR(s): #36526
  - File  PhysicsTools/NanoAOD/python/photons_cff.py  modified in PR(s): #36845, #36526

cmsbuild commented:
Pull request #36037 was updated. @cmsbuild, @mariadalfonso, @gouskos, @fgolf can you please check and sign again.

mariadalfonso commented:
please test

cmsbuild commented:
+1

**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-7fdc3b/22349/summary.html
**COMMIT**: c31707319cab81cdfc2c653b70feec6613dc4921
**CMSSW**: CMSSW_12_3_X_2022-02-10-2300/slc7_amd64_gcc10
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week1/cms-sw/cmssw/36037/22349/install.sh` to create a dev area with all the needed externals and cmssw changes.



The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
 - @adewit cms-sw/cmssw#36660
 - @bsunanda cms-sw/cmssw#36912
 - @makortel cms-sw/cmssw#36923

You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-7fdc3b/22349/git-recent-commits.json
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-7fdc3b/22349/git-merge-result


## Comparison Summary



**Summary**:
 - No significant changes to the logs found
 - ROOTFileChecks: Some differences in event products or their sizes found
 - Reco comparison results: 12 differences found in the comparisons
 - DQMHistoTests: Total files compared: 46
 - DQMHistoTests: Total histograms compared: 3764395
 - DQMHistoTests: Total failures: 2
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 3764371
 - DQMHistoTests: Total skipped: 22
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 7.296 KiB( 45 files compared)
 - DQMHistoSizes: changed ( 11634.0,... ): 0.912 KiB        Physics/NanoAODDQM
 - Checked 193 log files, 42 edm output root files, 46 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_12_3_X_2022-02-10-2300+7fdc3b/48322/triggerResults): no differences found

mariadalfonso commented:
+xpog

changes as expected on BeanSpot and Rho

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next master IBs (tests are also fine). This pull request will now be reviewed by the release team before it's merged. @perrotta, @dpiparo, @qliphy (and backports should be raised in the release meeting by the corresponding L2)

perrotta commented:
+1
- New variables added to nanoAod output
